### PR TITLE
Azure configuration source replacement

### PR DESCRIPTION
**Pull request by yvesgoeleven**
Closes https://github.com/Particular/NServiceBus.Host.AzureCloudService/issues/88

Documents how users can achieve behavior similar to the original azure configuration source. How to retrieve config settings from either roleenvironment if available, or from xml config otherwise.

### CODE REVIEW

**Comment by yvesgoeleven** [comment]
@Particular/azure-maintainers ready for review

File: Snippets/CloudServicesHost/CloudServicesHost_8/AzureConfigurationSourceReplacement.cs [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Configuration;
+using System.Web.Configuration;
+using System.Web.Hosting;
+
+class AzureConfigurationSourceReplacement
+{
+    AzureConfigurationSourceReplacement(EndpointConfiguration endpointConfiguration)
```
Comment (SeanFeldman) [inline Snippets/CloudServicesHost/CloudServicesHost_8/AzureConfigurationSourceReplacement.cs]: `endpointConfiguration` is not used, remove?

File: Snippets/CloudServicesHost/CloudServicesHost_8/AzureConfigurationSourceReplacement.cs [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    }
+}
+
+#region configuration-resolver
+
+static class ConfigurationResolver
+{
+    public static Configuration GetConfigurationHandler()
+    {
+        if (IsWebsite()) return WebConfigurationManager.OpenWebConfiguration(HostingEnvironment.ApplicationVirtualPath);
```
Comment (SeanFeldman) [inline Snippets/CloudServicesHost/CloudServicesHost_8/AzureConfigurationSourceReplacement.cs]: Condition on the same line with execution? Let's do it right, on separate lines. 🙂

File: Snippets/CloudServicesHost/CloudServicesHost_8/AzureConfigurationSourceReplacement.cs (hunk 1 above)
Comment (SeanFeldman) [reply Snippets/CloudServicesHost/CloudServicesHost_8/AzureConfigurationSourceReplacement.cs]: Hmmm, I think a snippet could be something like 

```c#
string GetConfigurationSection(string sectionName, string attributeName)
{
  // awesome code you provided
  return value;
}
```

I can guarantee people just going to copy/paste and use it as-is. So why to force them to create their own methods and go an extra step just to re-use the code? 🙂

File: Snippets/CloudServicesHost/CloudServicesHost_8/AzureConfigurationSourceReplacement.cs [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+
+        return ConfigurationManager.OpenExeConfiguration(ConfigurationUserLevel.None);
+    }
+
+    static bool IsWebsite()
+    {
+        return HostingEnvironment.IsHosted;
```
Comment (SeanFeldman) [inline Snippets/CloudServicesHost/CloudServicesHost_8/AzureConfigurationSourceReplacement.cs]: good candidate for expression body

**Pull request merged**